app/routers/event.py

```diff
@@ -70,65 +76,98 @@ async def create_event_api(event: EventModel, session=Depends(get_db)):
-async def create_new_event(request: Request,
-                           session=Depends(get_db)) -> Response:
+@router.post('/edit', include_in_schema=False)
+async def create_new_event(
+    request: Request, event_img: bytes = File(None), session=Depends(get_db)
+) -> Response:
     data = await request.form()
     title = data['title']
     content = data['description']
-    start = dt.strptime(data['start_date'] + ' ' +
-                        data['start_time'], TIME_FORMAT)
-    end = dt.strptime(data['end_date'] + ' ' + data['end_time'],
-                      TIME_FORMAT)
+    start = dt.strptime(data['start_date'] + ' ' + data['start_time'],
+                        TIME_FORMAT)
+    end = dt.strptime(data['end_date'] + ' ' + data['end_time'], TIME_FORMAT)
     owner_id = get_current_user(session).id
     availability = data.get('availability', 'True') == 'True'
     location = data['location']
     vc_link = data['vc_link']
     category_id = data.get('category_id')
 
     invited_emails = get_invited_emails(data['invited'])
-    uninvited_contacts = get_uninvited_regular_emails(session, owner_id,
-                                                      title, invited_emails)
+    uninvited_contacts = get_uninvited_regular_emails(session, owner_id, title,
+                                                      invited_emails)
 
     if vc_link is not None:
         raise_if_zoom_link_invalid(vc_link)
 
-    event = create_event(db=session, title=title, start=start, end=end,
-                         owner_id=owner_id, content=content,
-                         location=location, vc_link=vc_link,
+    event = create_event(db=session,
+                         title=title,
+                         start=start,
+                         end=end,
+                         owner_id=owner_id,
+                         content=content,
+                         location=location,
+                         vc_link=vc_link,
                          invitees=invited_emails,
                          category_id=category_id,
                          availability=availability)
 
+    if event_img:
+        image = process_image(event_img, event.id)
+        event.image = image
+        session.commit()
+
     messages = get_messages(session, event, uninvited_contacts)
-    return RedirectResponse(router.url_path_for('eventview', event_id=event.id)
-                            + f'messages={"---".join(messages)}',
-                            status_code=status.HTTP_302_FOUND)
+    return RedirectResponse(
+        router.url_path_for('eventview', event_id=event.id) +
+        f'?messages={"---".join(messages)}',
+        status_code=status.HTTP_302_FOUND)
+
+
+def process_image(img: bytes, event_id: int, img_height: int = HEIGHT) -> str:
+    """Resized and saves picture without exif (to avoid malicious date))
+    according to required height and keep aspect ratio"""
+    image = Image.open(io.BytesIO(img))
```

Comment:
I think we should first check if it is actually an image.

Reply:
I added a try: and except IOerror for this.